### PR TITLE
Protective-Devil 128

### PR DESCRIPTION
**Pull request by TrevorJamesH**

### CODE REVIEW

File: spec/binarySearchTree.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  context('search()', () => {
+    it('returns a node object.', () => {
+      const bst = new BinarySearchTree()
+
+      bst.search(10)
+      bst.search(15)
+      bst.search(20)
+
+      expect(bst.count).to.eql(0)
+    })
```
Comment (ai-santos) [inline spec/binarySearchTree.js:35]: the tests for lines 26-35 are not clear. Can you run the bst.search function inside of the expect and see if it returns that node?

File: lib/queue.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
+'use strict';
```
Comment (deonna) [inline lib/queue.js:1]: Consider adding `lib/` to your `.gitignore` to reduce the clutter of including transpiled files in your repo.

File: spec/binarySearchTree.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+  context('insert()', () => {
+    it('inserts a node with the specified value into the tree.', () => {
+      const bst = new BinarySearchTree()
+
+      bst.insert(1)
+      bst.insert(2)
+      bst.insert(3)
+
+      expect(bst.count).to.eql(2)
```
Comment (deonna) [inline spec/binarySearchTree.js:22]: Why should `bst.count` equal 2 after three elements are inserted?

Comment (deonna) [reply spec/binarySearchTree.js:22]: `eql` is for deep equality comparisons (e.g., comparing an array to another array). Use `equal` here.

Also: `eql` is poorly named by Chai. Use the alternative `to.deep.equal` instead.

File: spec/binarySearchTree.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      bst.search(15)
+      bst.search(20)
+
+      expect(bst.count).to.eql(0)
+    })
+
+    it('returns null if node is not found.', () => {
+      const bst = new BinarySearchTree()
+
+      expect(bst.count).to.eql(0)
```
Comment (deonna) [inline spec/binarySearchTree.js:40]: This test does not line up with the description.

I would expect that you would be instantiating a new BST (which is what you did on line 38), then attempting to search for a node that is not yet in the BST, then testing if the result of the search is `null` in your assertion.

File: spec/binarySearchTree.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    })
+  })
+
+  context('remove()', () => {
+    it('removes an value\'s node from the tree.', () => {
+      const bst = new BinarySearchTree()
+
+      bst.remove(20)
```
Comment (deonna) [inline spec/binarySearchTree.js:48]: The BST needs to have contents before values are removed from it. You instantiate it on line 46, but don't insert any nodes before removing nodes, so it is expected that `bst.count` will equal 0 even if `remove()` does not work.

**Pull request closed without merging**